src/EditorFeatures/Core/EditorConfigSettings/DataProvider/SettingsProviderBase.cs

```diff
@@ -131,6 +131,34 @@ public override bool TryGetValue(string key, [NotNullWhen(true)] out string? val
                 value = null;
                 return false;
             }
+
+            public override IEnumerable<string> Keys
+            {
+                get
+                {
+                    foreach (var key in _workspaceOptions.Keys)
+                        yield return key;
+
+                    if (!_result.HasValue)
+                        yield break;
+
+                    foreach (var key in _result.Value.AnalyzerOptions.Keys)
```

Comment:
❔ Can this end up returning the same key more than once? This would not be expected for an editorconfig parser.

Reply:
It shouldn't. AnalyzerOptions is a dictionary, so the keys are unique. We also check if we already returned the key above by checking `_workspaceOptions.TryGetValue`.

Reply:
I'll add comment on `Keys` property that it should return unique keys.